packages/drivers/routerlicious-driver/src/test/r11sSocketTests.spec.ts

```diff
@@ -0,0 +1,226 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { IClient } from "@fluidframework/driver-definitions";
+import {
+	DriverErrorTypes,
+	IResolvedUrl,
+	type IAnyDriverError,
+} from "@fluidframework/driver-definitions/internal";
+import { stub } from "sinon";
+import { Socket } from "socket.io-client";
+
+import { DefaultTokenProvider } from "../defaultTokenProvider.js";
+import { DocumentService } from "../documentService.js";
+import { RouterliciousDocumentServiceFactory } from "../documentServiceFactory.js";
+import {
+	R11sServiceClusterDrainingErrorCode,
+	RouterliciousErrorTypes,
+} from "../errorUtils.js";
+import * as socketModule from "../socketModule.js";
+// eslint-disable-next-line import/no-internal-modules
+import { ClientSocketMock } from "../test/socketTestUtils.ts/socketMock.js";
+
+describe("R11s Socket Tests", () => {
+	let documentService: DocumentService;
+	let routerliciousDocumentServiceFactory: RouterliciousDocumentServiceFactory;
+	// let deltaConnection: R11sDocumentDeltaConnection;
+	let resolvedUrl: IResolvedUrl;
+	let socket: ClientSocketMock | undefined;
+
+	async function mockSocket<T>(_response: Socket, callback: () => Promise<T>): Promise<T> {
+		const getSocketCreationStub = stub(socketModule, "SocketIOClientStatic");
+		getSocketCreationStub.returns(_response);
+		try {
+			return await callback();
+		} finally {
+			getSocketCreationStub.restore();
+		}
+	}
+
+	const client: IClient = {
+		mode: "read",
+		details: { capabilities: { interactive: true } },
+		permission: [],
+		user: { id: "id" },
+		scopes: [],
+	};
+
+	beforeEach(async () => {
+		routerliciousDocumentServiceFactory = new RouterliciousDocumentServiceFactory(
+			new DefaultTokenProvider("jwt"),
+		);
+		resolvedUrl = {
+			type: "fluid",
+			id: "id",
+			url: "https://mock.url/tenantId/documentId",
+			endpoints: {
+				ordererUrl: "ordererUrl",
+				storageUrl: "storageUrl",
+				deltaStorageUrl: "deltaStorageUrl",
+				deltaStreamUrl: "deltaStreamUrl",
+			},
+			tokens: {},
+		};
+		documentService = (await routerliciousDocumentServiceFactory.createDocumentService(
+			resolvedUrl,
+		)) as DocumentService;
+	});
+
+	it("connect_document_error with Token Revoked error", async () => {
+		const errorToThrow = {
+			code: 403,
+			message: "TokenRevokedError",
+			retryAfterMs: 10,
+			internalErrorCode: "TokenRevoked",
+			errorType: DriverErrorTypes.authorizationError,
+			canRetry: false,
+		};
+		const errorEventName = "connect_document_error";
+		socket = new ClientSocketMock({
+			connect_document: { eventToEmit: errorEventName, errorToThrow },
+		});
+
+		let errorOccurred = false;
+		try {
+			await mockSocket(socket as unknown as Socket, async () =>
```

Comment:
nit: could use `await assert.rejects()` here instead of the try..catch 
https://nodejs.org/api/assert.html#assertrejectsasyncfn-error-message

Reply:
done